migrator/migrations/m_202_to_m_203_vuln_requests_for_suppressed_cves/migration_impl.go

```diff
@@ -0,0 +1,280 @@
+	}
+
+	log.Infof("Found %d globally snoozed vulnerabilities that need to be migrated", count)
+	return cveMap, nil
+}
+
+func createVulnRequests(ctx context.Context, database *types.Databases, now *protocompat.Timestamp, cveMap map[string]*protocompat.Timestamp) error {
+	store := vulnReqStore.New(database.PostgresDB)
+
+	var vulnReqs []*storage.VulnerabilityRequest
+	var count int
+	for cve, expiry := range cveMap {
+		// If the snooze expiry is past due, no need to create v2 exceptions since those will be reverted on Central startup anyway.
+		if expiry != nil && protocompat.CompareTimestamps(now, expiry) >= 0 {
+			continue
+		}
+
+		// We do not allow multiple exceptions covering the same scope.
+		// The following check will return true in case of upgrade-rollback-upgrade path either because of:
+		// - previous migration run, or
+		// - upgrade -> v2 request created -> rollback -> same v1 snooze created -> upgrade
+		matchingReqsExist, err := checkMatchingRequestsExist(ctx, database, cve)
+		if err != nil {
+			return errors.Wrapf(err, "failed to check if global vulnerability exception for CVE %s exist in the database", cve)
+		}
+		if matchingReqsExist {
+			log.Infof(fmt.Sprintf("global vulnerability exception already exists for CVE %s", cve))
+			continue
+		}
+
+		vulnReqs = append(vulnReqs, createVulnerabilityRequest(cve, now, expiry))
+		count++
+
+		if len(vulnReqs) == batchSize {
+			if err := store.UpsertMany(ctx, vulnReqs); err != nil {
+				return errors.Wrapf(err, "failed to upsert %d vulnerability exceptions after %d were upserted", len(vulnReqs), count-len(vulnReqs))
+			}
+			vulnReqs = vulnReqs[:0]
+		}
+	}
+
+	if len(vulnReqs) > 0 {
+		if err := store.UpsertMany(ctx, vulnReqs); err != nil {
+			return errors.Wrapf(err, "failed to upsert last %d objects", len(vulnReqs))
+		}
+	}
+	log.Infof("Created %d vulnerability exceptions for globally snoozed vulnerabilities", count)
+	return nil
+}
+
+func updateImageCVEEdges(ctx context.Context, database *types.Databases, cveMap map[string]*protocompat.Timestamp) error {
+	cves := make([]string, 0, len(cveMap))
+	for cve := range cveMap {
+		cves = append(cves, cve)
+	}
+
+	db := database.GormDB.WithContext(ctx).Table(schema.ImageCveEdgesTableName)
+	query := database.GormDB.WithContext(ctx).Table(schema.ImageCveEdgesTableName).
+		Raw(fmt.Sprintf("SELECT %[1]s.serialized FROM %[1]s WHERE EXISTS (SELECT 1 FROM %[2]s WHERE %[1]s.imagecveid = %[2]s.id AND %[2]s.cvebaseinfo_cve = ANY($1::text[]))", schema.ImageCveEdgesTableName, schema.ImageCvesTableName), cves)
+	rows, err := query.Rows()
+	if err != nil {
+		return errors.Wrapf(err, "failed to query table %s", schema.ImageCvesTableName)
+	}
+	defer func() { _ = rows.Close() }()
+
+	var convertedEdges []*schema.ImageCveEdges
+	var count int
+	for rows.Next() {
+		var obj schema.ImageCveEdges
+		if err = query.ScanRows(rows, &obj); err != nil {
```

Comment:
Shouldn't we scope the `err` to the if since we are using it in an if?
```suggestion
		if err := query.ScanRows(rows, &obj); err != nil {
```

Reply:
^ The difference between the `=` and `:=` reminds me of JavaScript. 🧌